### PR TITLE
atlasexec: added exec-order argument

### PR DESCRIPTION
**Pull request by giautm**

### CODE REVIEW

File: atlasexec/atlas.go [hunk 1]
```diff
@@ -133,6 +137,13 @@ const (
 	TriggerTypeGithubAction TriggerType = "GITHUB_ACTION"
 )
 
+// ExecutionOrder values.
+const (
+	ExecutionOrderLinear     ExecutionOrder = "linear" // Default
+	ExecutionOrderLinearSkip ExecutionOrder = "linear-skip"
+	ExecutionOrderNonLinear  ExecutionOrder = "non-linear"
```
Comment (a8m) [inline atlasexec/atlas.go:144]:
```suggestion
	ExecOrderLinear     MigrateExecOrder = "linear" // Default
	ExecOrderLinearSkip MigrateExecOrder = "linear-skip"
	ExecOrderNonLinear  MigrateExecOrder = "non-linear"
```

File: atlasexec/atlas.go [hunk 2]
```diff
@@ -39,6 +39,9 @@ type (
 	}
 	// TriggerType defines the type for the "trigger_type" enum field.
 	TriggerType string
+	// ExecutionOrder define how Atlas computes and executes pending migration files to the database.
+	// See: https://atlasgo.io/versioned/apply#execution-order
+	ExecutionOrder string
```
Comment (a8m) [inline atlasexec/atlas.go:44]:
```suggestion
	MigrateExecOrder string
```

**Review by a8m — APPROVED** [review]

File: atlasexec/atlas_test.go [hunk 3]
```diff
@@ -759,3 +759,49 @@ func sqlitedb(t *testing.T) string {
...
+			params: &atlasexec.MigrateApplyParams{
+				ExecOrder: atlasexec.ExecutionOrderLinear,
+			},
+			expect: "migrate apply --format {{ json . }} --exec-order linear",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := c.MigrateApply(context.Background(), tt.params)
+			require.Error(t, err)
+			require.Equal(t, tt.expect, err.Error())
```
Comment (ronenlu) [inline atlasexec/atlas_test.go:804]: why the error should be equal to expected cmd: `migrate apply --format {{ json . }} --env test` ?

Comment (giautm) [reply atlasexec/atlas_test.go:804]: kk, that's a trick to get all output from our mock-args script. ;)

Comment (giautm) [reply atlasexec/atlas_test.go:804]: I will add a comment to explain,

**Pull request merged**